main.py

```diff
@@ -140,3 +160,102 @@ def update_circuits(self, event):
         # pylint: disable=unused-argument
         if settings.FIND_CIRCUITS_IN_FLOWS:
             self.automate.find_circuits()
+
+    @staticmethod
+    def get_stored_flows(dpids: list = None, state: str = None):
+        """Get stored flows from flow_manager napps."""
+        api_url = f'{settings.FLOW_MANAGER_URL}/stored_flows'
+        if dpids:
+            str_dpids = ''
+            for dpid in dpids:
+                str_dpids += f'&dpid={dpid}'
+            api_url += '/?'+str_dpids[1:]
+        if state:
+            char = '&' if dpids else '/?'
+            api_url += char+f'state={state}'
+        result = requests.get(api_url)
+        flows_from_manager = result.json()
+        return flows_from_manager
+
+    @classmethod
+    def do_match(cls, flow, args):
+        """Match a packet against this flow (OF1.3)."""
+        # pylint: disable=consider-using-dict-items
+        if ('match' not in flow['flow']) or (len(flow['flow']['match']) == 0):
+            return False
+        for name in flow['flow']['match']:
+            field_flow = flow['flow']['match'][name]
+            if name not in args:
+                return False
+            if name == 'dl_vlan':
+                field = args[name][-1]
+            else:
+                field = args[name]
+            if name not in ('ipv4_src', 'ipv4_dst', 'ipv6_src', 'ipv6_dst'):
+                if field_flow != field:
+                    return False
+            else:
+                packet_ip = int(ipaddress.ip_address(field))
+                ip_addr = flow['flow']['match'][name]
+                if packet_ip & ip_addr.netmask != ip_addr.address:
+                    return False
+        return flow
+
+    def match_flows(self, switch, args, many=True):
+        # pylint: disable=bad-staticmethod-argument
+        """
+        Match the packet in request against the stored flows from flow_manager.
+        Try the match with each flow, in other. If many is True, tries the
+        match with all flows, if False, tries until the first match.
+        :param args: packet data
+        :param many: Boolean, indicating whether to continue after matching the
+                first flow or not
+        :return: If many, the list of matched flows, or the matched flow
+        """
+        stored_flows = self.get_stored_flows()
```

Comment:
`match_flows` is called iterativelly, and every time you're refetching all stored_flows from all dpids, so to be aligned with the strategy of bulk request, let's pre-fetch all the stored flows and pass them and then you iterate on them. We need this endpoint to perform well and be optimized since mef_eline will rely on it to run sdntrace for hundreds of EVCs periodically.

Reply:
Fine @viniarck, that's why before I used the self.store_flows variable that I updated every time the endpoint was called. Now I have done it with a variable that is passed by parameter.

Reply:
Vinicius, thanks, you are right.